imputation/glimpse_hail_batch/glimpse_hail_batch/imputation_jobs.py

```diff
@@ -0,0 +1,335 @@
+        j.command(f'gcloud storage rm {sample_group.temp_dir}/phase/*/*/*')
+        j.command(f'gcloud storage rm {sample_group.temp_dir}/phase/*/*')
+        j.command(f'gcloud storage rm {sample_group.temp_dir}/phase/*')
+        j.command(f'gcloud storage rm {sample_group.temp_dir}/ligate/*')
+
+    return j
+
+
+def merge_vcfs(b: hb.Batch,
+               sample_group: SampleGroup,
+               file_path_regex: str,
+               output_path: str,
+               docker: str,
+               cpu: int,
+               memory: str,
+               storage: str,
+               use_checkpoint: bool) -> Optional[Job]:
+    if use_checkpoint and hfs.exists(output_path):
+        return None
+
+    j = b.new_bash_job(attributes={'name': f'merge-vcfs/sample-group-{sample_group.sample_group_index}',
+                                   'task': 'merge-vcfs'})
+    j.cpu(cpu)
+    j.image(docker)
+    j.storage(storage)
+    j.memory(memory)
+
+    j.command(f'''
+python3 << EOF
+import hail as hl
+hl.init(backend="spark", local="local[{cpu}]", default_reference="GRCh38", tmp_dir="/io/", local_tmpdir="/io/")
+mt = hl.import_vcf("{file_path_regex}")
+mt = mt.annotate_rows(info=mt.info.annotate(N={len(sample_group.samples)}, AF=mt.info.AF[0], INFO=mt.info.INFO[0], RAF=mt.info.RAF[0]))
+mt.write("{output_path}", overwrite=True)
+EOF    
+''')
+
+    return j
+
+
+def write_success(b: hb.Batch, sample_group: SampleGroup, docker: str) -> Job:
+    j = b.new_bash_job(attributes={'name': f'write-success/sample-group-{sample_group.sample_group_index}',
+                                   'task': 'write-success'})
+    j.cpu(1)
+    j.image(docker)
+    j.command(f'''
+python3 << EOF
+import hailtop.fs as hfs
+with hfs.open("{sample_group.success_file}", "w") as f:
+    f.write("0")
+EOF
+''')
+
+    return j
+
+
+def union_sample_groups(b: hb.Batch,
+                        mt_paths: hb.ResourceFile,
+                        output_path: str,
+                        docker: str,
+                        cpu: int,
+                        memory: str,
+                        storage: str,
+                        batch_name: str,
+                        billing_project: str,
+                        remote_tmpdir: str,
+                        regions: str,
+                        use_checkpoints: bool) -> Optional[Job]:
+    if use_checkpoints and hfs.exists(output_path):
+        return None
+
+    j = b.new_bash_job(attributes={'name': 'union_sample_groups'})
+    j.cpu(cpu)
+    j.image(docker)
+    j.storage(storage)
+    j.memory(memory)
+
+    cmd = f"""
+hailctl config set batch/billing_project "{billing_project}"
+hailctl config set batch/remote_tmpdir "{remote_tmpdir}"
+hailctl config set batch/regions "{','.join(regions)}"
+
+python3 << EOF
+import hail as hl
+import os
+from typing import List
+
+hl.init(backend='batch', app_name='{batch_name}-union-sample-groups', driver_cores=16, worker_cores=2)
```

Comment:
We need confirmation from the Hail team that these are suitable parameters for driver and worker cores.